File: clmanager.cs
using System;
using System.IO;
using System.Collections.Generic;

class checklist
{
	static void writeln(string message) { Console.WriteLine(message); }
	static void writeln(string format, params object[] args) { Console.WriteLine(format, args); }
	static string readln() { return Console.In.ReadLine(); }

	public void add_checklist(){
		string	name;
		string	file_name;

		writeln("Please enter a name for the list (if empty, default name will be assigned): ");
		name = readln();

		if (string.IsNullOrEmpty(name))
		{
			name = "Default";
		}

		writeln("Please enter a file name for the list (if blank, filename will default to lists name): ");
		file_name = readln();
		if (string.IsNullOrEmpty(file_name) || file_name == ".txt"){
			file_name = name + ".txt";
		}
		else if (file_name.IndexOf(".txt") < 0){
			file_name += ".txt";
		} 
		
		// Assures there's no objects with duplicate names in the list
		while (check_duplicate(name)){
			writeln("{0} already exists in the list, adjusting to fit in list.", name);
			name = "_" + name;
		}
		clists.Add(name, new clist(name, file_name));
	}

	// Checks for any duplicates in the list.
	bool check_duplicate(string search_key){
		return clists.ContainsKey(search_key);
	}

	public void print_lists()
	{
		foreach (var list in clists.Values)
		{
			list.print_list();
		}
	}

	SortedDictionary<string, clist> clists = new SortedDictionary<string, clist>();
}

File: clmanager.cpp
#include "clmanager.h"
#include <vector>
#include <string>
#include <iostream>

using namespace std;

void checklist::add_checklist(){
	char	input;
	string	name;
	string	file_name;

	cout << "Please enter a name for the list (if empty, default name will be assigned): " << endl;
	getline(cin, name);
	if (name.empty()){
		name = "Default";
	}

	cout << "Please enter a file name for the list (if blank, filename will default to lists name): " << endl;
	getline(cin, file_name);
	if (file_name.empty() || file_name == ".txt"){
		file_name = name + ".txt";
	}
	else if (!file_name.find(".txt")){
		file_name += ".txt";
	}

	// Assures there's no objects with duplicate names in the list
	while (check_duplicate(name)){
		cout << name << " already exists in the list, adjusting to fit in list." << endl;
		name = "_" + name;
	}
	clists.push_back(new clist(name, file_name));
}

void checklist::print_lists(){
	for (auto index : clists){
		index->print_list();
	}
}

bool checklist::check_duplicate(string search_key){
	for (auto i : clists){
		if (i->get_name() == search_key){
			return true;
		}
	}
	return false;
}